packages/firebase_ml_vision/lib/src/barcode_detector.dart

```diff
@@ -4,18 +4,599 @@
-    // TODO: implement detectInImage
+  Future<List<VisionBarcode>> detectInImage(
+      FirebaseVisionImage visionImage) async {
+    final List<dynamic> reply = await FirebaseVision.channel.invokeMethod(
+      'BarcodeDetector#detectInImage',
+      visionImage.imageFile.path,
+    );
+
+    final List<VisionBarcode> barcodes = <VisionBarcode>[];
+    reply.forEach((dynamic barcode) {
+      barcodes.add(new VisionBarcode._(barcode));
+    });
+
+    return barcodes;
   }
 }
 
-class BarcodeDetectorOptions {}
+/// Represents a single recognized barcode and its value.
+class VisionBarcode {
+  VisionBarcode._(Map<dynamic, dynamic> _data)
+      : boundingBox = _data['left'] != null
+            ? Rectangle<int>(
+                _data['left'],
+                _data['top'],
+                _data['width'],
+                _data['height'],
+              )
+            : null,
+        rawValue = _data['raw_value'],
+        displayValue = _data['display_value'],
+        format = VisionBarcodeFormat._(_data['format']),
+        cornerPoints = _data['points'] == null
+            ? null
+            : _data['points']
+                .map<Point<int>>((dynamic item) => Point<int>(
+                      item[0],
+                      item[1],
+                    ))
+                .toList(),
+        valueType =
+            VisionBarcodeValueType.values.elementAt(_data['value_type']),
+        email = _data['email'] == null
+            ? null
+            : VisionBarcodeEmail._(_data['email']),
+        phone = _data['phone'] == null
+            ? null
+            : VisionBarcodePhone._(_data['phone']),
+        sms = _data['sms'] == null ? null : VisionBarcodeSMS._(_data['sms']),
+        url = _data['url'] == null
+            ? null
+            : VisionBarcodeURLBookmark._(_data['url']),
+        wifi =
+            _data['wifi'] == null ? null : VisionBarcodeWiFi._(_data['wifi']),
+        geoPoint = _data['geo_point'] == null
+            ? null
+            : VisionBarcodeGeoPoint._(_data['geo_point']),
+        contactInfo = _data['contact_info'] == null
+            ? null
+            : VisionBarcodeContactInfo._(_data['contact_info']),
+        calendarEvent = _data['calendar_event'] == null
+            ? null
+            : VisionBarcodeCalendarEvent._(_data['calendar_event']),
+        driverLicense = _data['driver_license'] == null
+            ? null
+            : VisionBarcodeDriverLicense._(_data['driver_license']);
+
+  /// Gets the bounding rectangle of the detected barcode.
+  ///
+  /// Returns null if the bounding rectangle can not be determined.
+  final Rectangle<int> boundingBox;
+
+  /// Returns barcode value as it was encoded in the barcode.
+  /// Structured values are not parsed, for example: 'MEBKM:TITLE:Google;URL://www.google.com;;'.
+  ///
+  /// Returns null if nothing found.
+  final String rawValue;
+
+  /// Returns barcode value in a user-friendly format.
```

Comment:
Since this is a variable, I believe it is better to say `The barcode value in a user-friendly format.` Returns implies that it is a method. Same for similar comments on variables below.

Reply:
You're right.